lib/carrierwave/orm/activerecord.rb

```diff
@@ -56,12 +56,15 @@ def mount_base(column, uploader=nil, options={}, &block)
       validates_processing_of column if uploader_option(column.to_sym, :validate_processing)
       validates_download_of column if uploader_option(column.to_sym, :validate_download)
 
+      after_save :"store_#{column}!"
       before_save :"write_#{column}_identifier"
-      after_save :"store_previous_changes_for_#{column}"
       after_commit :"remove_#{column}!", :on => :destroy
       after_commit :"mark_remove_#{column}_false", :on => :update
+
+      after_save :"store_previous_changes_for_#{column}"
+      after_commit :"reset_previous_changes_for_#{column}"
```

Comment:
Honestly, this confuses me. Is the order of callbacks the reverse of add order?

Reply:
[It appears so](https://api.rubyonrails.org/classes/ActiveSupport/Callbacks.html#method-i-run_callbacks): 

> Calls the before and around callbacks in the order they were set, yields the block (if given one), and then runs the after callbacks in reverse order.

Given that this is an after callback, they are executed in reverse.